src/registry/addressNL/initialValues.ts

```diff
@@ -0,0 +1,30 @@
+import type {AddressData, AddressNLComponentSchema} from '@open-formulieren/types';
+
+import type {GetInitialValues} from '@/registry/types';
+import type {JSONObject} from '@/types';
+
+const getInitialValues: GetInitialValues<AddressNLComponentSchema, JSONObject> = ({
```

Comment:
But can we directly say that the initialValues is of the type `AddressData`, right?

```suggestion
const getInitialValues: GetInitialValues<AddressNLComponentSchema, AddressData> = ({
```

This will require an update to the `RegistryEntry` type, which will have to allow a dynamic type for setting the `getInitialValues` return value

Reply:
the type checker doesn't like this because `AddressData` is not assignable to `JSONObject`, which the bigger registry machinery expects.

It did not think it's worth complicating the type checking so much for something that is comes down to some TS nuances. The intent we express is that something has to be a `JSONValue`, i.e. some that can go over the (API) wire as JSON data. The type definition for `JSONObject` / `JSONValue` that we now have is the best we can. `AddressData` fits the intent - it can be JSON serialized, but due to the missing `[string]` index type TS does not consider it valid for `JSONObject`.

Maybe using `{[k in keyof AddressData]: JSONValue}` would work, but I remember spending hours and hours on making these kind of things working in the type system and didn't want to burn my time on this, instead prefer focusing on getting to 1.0.